library/core/src/num/f32.rs

```diff
@@ -1034,6 +1034,7 @@ impl f32 {
             }
         }
 
+        #[inline(always)]
```

Comment:
I don't think this is the right fix. I think the collector needs to handle these functions even without `#[inline(always)]`.

Reply:
The collector runs on `instance_mir` which runs on the shim mir or `optimized_mir`, which should already have the const_eval_select calls rewired.

Reply:
Yeah, no clue why it fails to work properly.

Reply:
I've tested with the arm target compiler and this is indeed a fix, so I guess this should be some ghost bug that people who add `const_eval_select` to library functions worry about in the future.

Reply:
I don't think that's a sufficiently satisfying answer; the bug is at minimum exposed by these changes. We should understand why (even if we don't necessarily fix it here) and track fixing it.

Reply:
The undefined reference is from compiler-builtins, which shouldn't have any link time dependencies on the core. We have been using inline to avoid this issue so far, so that seems fine to me, although personally I would prefer approach proposed in rust-lang/compiler-builtins#472.

Reply:
Right, forgot about that.

Reply:
I created an issue and linked to it from this attribute.